### PR TITLE
Migrate consumesMany() to edm::GetterOfProducts() in GlobalHitsAnalyzer

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description:
Based on PR #40167 . 
Validation code :
Migrate GlobalHitsAnalyzer away from consumesMany() to use edm::GetterOfProducts().

#### PR validation:
Tested in CMSSW_13_0_0_pre4, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

File: Validation/GlobalHits/src/GlobalHitsAnalyzer.cc [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include "Geometry/HcalCommonData/interface/HcalHitRelabeller.h"
 #include "Geometry/HcalTowerAlgo/interface/HcalGeometry.h"
 #include "Geometry/Records/interface/HcalRecNumberingRecord.h"
+#include "FWCore/Framework/interface/GetterOfProducts.h"
+#include "FWCore/Framework/interface/ProcessMatch.h"
+
+edm::GetterOfProducts<edm::HepMCProduct> getterOfProducts_;
```
Comment (missirol) [inline Validation/GlobalHits/src/GlobalHitsAnalyzer.cc:20]: Shouldn't this be a private member of the class `GlobalHitsAnalyzer` like, for example, `tGeomToken_` ?
https://github.com/cms-sw/cmssw/blob/d32bfd821fd0a23b7eebb354fc51b2f7917962b9/Validation/GlobalHits/interface/GlobalHitsAnalyzer.h#L147

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**